### PR TITLE
Added ZK storage to EP contracts for forward compatibility

### PR DESCRIPTION
**Pull request by Sidu28**

### CODE REVIEW

**Comment by wadealexc** [comment]
Is this ready for review now, or still WIP?

File: src/contracts/pods/EigenPod.sol [hunk 1]
```diff
@@ -93,6 +93,18 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
     /// @notice This variable tracks any ETH deposited into this contract via the `receive` fallback function
     uint256 public nonBeaconChainETHBalanceWei;
 
+     /// @notice This variable tracks the total amoutn of partial withdrawals claimed via merkle proofs prior to a switch to ZK proofs for claiming partial withdrawals
+    uint64 public totalPartialWithdrawalAmountClaimedGwei;
```
Comment (wadealexc) [inline src/contracts/pods/EigenPod.sol:97]: This name is word soup. Can we do something shorter?

How about `sumPartialWithdrawalsClaimedGwei`

If we can get something shorter that's dope too

**Comment by wadealexc** [comment]
Why are we adding unused variables?

**Comment by Sidu28** [comment]
> Why are we adding unused variables?

to help with forward compatibility, ie, storage already exists for when we upgrade implementation to ZK proofs

**Comment by ChaoticWalrus** [comment]
> > Why are we adding unused variables?
> 
> to help with forward compatibility, ie, storage already exists for when we upgrade implementation to ZK proofs

We should not be adding purely unused storage variables. This PR should cover logic + storage changes that are _necessary_ for forwards compatibility, and exclude adding totally unused storage variables that you think you might be added later.

File: src/contracts/pods/EigenPod.sol [hunk 2]
```diff
@@ -93,6 +93,9 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
     /// @notice This variable tracks any ETH deposited into this contract via the `receive` fallback function
     uint256 public nonBeaconChainETHBalanceWei;
 
+     /// @notice This variable tracks the total amoutn of partial withdrawals claimed via merkle proofs prior to a switch to ZK proofs for claiming partial withdrawals
```
Comment (ChaoticWalrus) [inline src/contracts/pods/EigenPod.sol:96]: typo: `amoutn` => `amount`

**Review by ChaoticWalrus — APPROVED** [review]
LGTM. noted one minor typo in a comment.

**Comment by Sidu28** [comment]
> > > Why are we adding unused variables?
> > 
> > 
> > to help with forward compatibility, ie, storage already exists for when we upgrade implementation to ZK proofs
> 
> We should not be adding purely unused storage variables. This PR should cover logic + storage changes that are _necessary_ for forwards compatibility, and exclude adding totally unused storage variables that you think you might be added later.

removed the extraneous stuff, take a look now

File: src/contracts/pods/EigenPod.sol (hunk 2 above)
Comment (Sidu28) [reply src/contracts/pods/EigenPod.sol]: fixed

**Pull request merged**